airflow/task_runner/base_task_runner.py

```diff
@@ -89,7 +90,7 @@ def __init__(self, local_task_job):
 
     def _read_task_logs(self, stream):
         while True:
-            line = stream.readline().decode('utf-8')
+            line = stream.readline().decode(sys.getdefaultencoding(), 'ignore')
```

Comment:
It seems to me that the 'ignore' parameter is resolving the issue rather than sys.getdefaultencoding()?